### ISSUE TITLE
Switch to jade-parser etc.

### ISSUE THREAD

**Issue by TimothyGu** [jadejs/then-jade#17]

**Comment by jeromew** [comment]
@TimothyGu currently `then-jade` is bound on jade 1.3.0

My objective is to reach a state-of-the-art jade dependency while avoiding manual upgrades because the monkey-patching is a but cumbersome.

When the jade 2.x project started, I told to myself that I would 
 * wait until 2.x was released,
 * do a last `then-jade` dependency update on the last 1.x
 * and then adopt the new modular architecture

when you say "switch to jade-parser" etc, do you mean (I haven't checked) that this module split has already been done in the 1.x branch (not waiting for 2.x) ?

If yes, can you tell me what was the last jade version with the old monolithic approach ? I would then updgrade then-jade to this version and start thinking about the effects of the modular design on `then-jade`.

**Comment by TimothyGu** [comment]
> when you say "switch to jade-parser" etc, do you mean (I haven't checked) that this module split has already been done in the 1.x branch (not waiting for 2.x) ?

No, I'm just raising awareness that at some point we should switch.

Thanks!

**Comment by jeromew** [comment]
`then-jade` is now upgraded to jade 1.11.0 (hopefully the last jade release on the 1.x release path)
I haven't done any research regarding the migration to the new architecture that is envisioned for the upcoming jade 2.0 release

**Comment by jonathanong** [comment]
i would like to know: is there an intent to migrate to the pugjs v2 architecture? not sure if i should bother using this while i migrate to pug

**Comment by jeromew** [comment]
check https://github.com/pugjs/pug/issues/2708 for news on this.

I published https://www.npmjs.com/package/then-pug which is a working implementation based on pug head (I use this and don't use then-jade anymore)

There is still work before the babel AST based code-gen lands into pug master so until then I think the project needs to be considered experimental.
